### DISCUSSION TITLE
Plugin Translation | Strings pending on Translate Wordpress

### DISCUSSION THREAD

**Discussion by Henrylq** [code4recovery/12-step-meeting-list#1401] [Q&A] [answered]
Hey there,

I recently wanted to use your plugin on a website for a local group that would definitely benefit from having it, I'm helping them remake the website. The first thing I noticed was that I needed to translate it, so I did on the [Translate Wordpress page](https://translate.wordpress.org/projects/wp-plugins/12-step-meeting-list/), actually went over all of the strings in the **Slovak** language and translated it. But they don't seem to move along to the next step of being accepted.

I would love to ask you if there are other steps I could take to get them approved and live as obviously this is a requirement for a website that's local. Of course I would be open to take other steps first if there's a need to do that, but I did my best to get as close as possible.

Please let me know! 🙂

**Comment by gkovats** [comment]
Hey Henry, that's awesome! I don't personally have access to see your submission, but I'll check with folks that do and see about including in a patch when we can. We'll get back to you in the next or so.

**Reply by gkovats** [reply, marked as answer]
Looking more into this, the Slovak page shows:
https://translate.wordpress.org/locale/sk/default/wp-plugins/12-step-meeting-list/
> The initial language pack for the plugin will be generated when 90% of the Stable (latest release) sub-project strings have been translated (currently 6%).

It appears basically this translation process is managed by a volunteer team outside of us. Basically, think these are the folks that can help with next steps:
https://make.wordpress.org/polyglots/teams/?locale=sk_SK

That said, an alternate approach is just download a PO and MO file from them here:
https://translate.wordpress.org/projects/wp-plugins/12-step-meeting-list/dev/sk/default/
![image](https://github.com/code4recovery/12-step-meeting-list/assets/1187583/53e82e94-7a5d-415e-b129-592ad4e1f422)

Then just copy the .po and .mo files over the
```
/wp-content/plugins/12-step-meeting-list/languages/12-step-meeting-list-sk_SK.mo
/wp-content/plugins/12-step-meeting-list/languages/12-step-meeting-list-sk_SK.po
```
That's _if_ you have access to your system files. I just tried it, and a good part of the plugin gets translated, but I can tell there's some areas we missed.

Let me know if this helps. I can't help with your submissions on translate.wordpress, but this middle step may get you going.

**Reply by Henrylq** [reply]
Awesome, this is exactly what I needed. I went with the alternative approach and will update strings when needed and just download these files into the plugin folder to update translations on the website I'm currently building for the local group's website in the future.

Thank you very much for the detailed step by step message, this helped a lot & does what I expected.

**Reply by Henrylq** [reply]
> I just tried it, and a good part of the plugin gets translated, but I can tell there's some areas we missed.

Regarding this, I noticed that actually quite a lot of important strings / parts of the plugin aren't translated doing it this way.

![image](https://github.com/code4recovery/12-step-meeting-list/assets/40896381/63ab4b10-f296-4cb5-b095-1ddcb1ae98bb)
![image](https://github.com/code4recovery/12-step-meeting-list/assets/40896381/0269d78c-6dfe-4617-9a21-221d8c7b47c4)

This is arguably the most important part of the UI and it's not translated / translatable doing it the way I did. Is there a way around this so that I can translate these parts as well, or could you add support for translating these / a way to translate this? I would very much appreciate that!

I would love if there was a way to translate everything without manually editing plugin files, so that I can keep auto-updates on & not mess with files I shouldn't really edit.

**Reply by gkovats** [reply]
We'll work on patching the phrases missing "gettext". Creating an issue from this thread to follow up with a fix.

**Reply by joshreisner** [reply]
hi @Henrylq that UI is actually part of a different project and we'd love to have your help translating it into slovak

it would involve translating this file:
https://github.com/code4recovery/tsml-ui/blob/main/src/i18n/en.ts by creating a new file called sk.ts

as well as adding sk keys to these files: https://github.com/code4recovery/spec/tree/main/data

thanks!

**Reply by joshreisner** [reply]
these changes would mean that visitors to any site that uses this ui will see the ui in slovak if that is their language preference, not just your site

**Reply by joshreisner** [reply]
thanks for your PRs @Henrylq - i'm seeing the changes now on live sites eg aasanjose.org/meetings

<img width="1319" alt="Screenshot 2024-03-29 at 9 30 34 AM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/4e8769fc-971e-469c-962e-0f34d562bc79">

**Reply by Henrylq** [reply]
That's awesome! Thank you for approving them so quickly & I'm very glad to see it localized. This will definitely be helpful for the local group and improve their website, helping people here. I'm glad I can use this project now to help others here.

Again, thank you very much for helping here! I think I'll close this discussion now as it definitely achieved what it was supposed to.
